### PR TITLE
test(atomic): added accessibility tests for all result components

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-971

### CODE REVIEW

File: packages/atomic/cypress/integration/common-assertions.ts [hunk 1]
```diff
@@ -10,9 +10,23 @@ export function should(should: boolean) {
   return should ? 'should' : 'should not';
 }
 
-export function assertAccessibility(component: string) {
+export function assertAccessibility(
+  component: string | (() => Cypress.Chainable<JQuery<HTMLElement>>)
+) {
+  const rulesToIgnore = ['landmark-one-main', 'page-has-heading-one', 'region'];
```
Comment (btaillon) [inline packages/atomic/cypress/integration/common-assertions.ts:16]: These three tests are relevant for a whole page, not a single component.

When we test accessibility on a single element by specifying a string as a parameter, they appear to already be skipped, but when passing an element they seem to be tested for some reason.

File: packages/atomic/cypress/integration/common-assertions.ts [hunk 2]
```diff
@@ -10,9 +10,23 @@ export function should(should: boolean) {
...
+  const rules = rulesToIgnore.reduce(
+    (obj, rule) => ({...obj, [rule]: {enabled: false}}),
+    {}
+  );
```
Comment (btaillon) [inline packages/atomic/cypress/integration/common-assertions.ts:20]: we don't have `fromEntries` in cypress tests

File: packages/atomic/cypress/integration/result-list/result-components/result-icon-selectors.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ export const resultIconComponent = 'atomic-result-icon';
 export const ResultIconSelectors = {
   shadow: () => cy.get(resultIconComponent).shadow(),
   firstInResult: () =>
-    ResultListSelectors.firstResult().find(resultIconComponent).shadow(),
```
Comment (btaillon) [inline packages/atomic/cypress/integration/result-list/result-components/result-icon-selectors.ts:8]: Some components had `.shadow()` on `firstInResult`, now none of them do.

File: packages/atomic/cypress/integration/result-list/result-list-utils.ts [hunk 4]
```diff
@@ -5,36 +5,46 @@ import {
...
+}
+
 export function withAnySectionnableResultList(
   assertions: (
     display: ResultDisplayLayout,
     imageSize: ResultDisplayImageSize,
     density: ResultDisplayDensity
-  ) => void
+  ) => void,
+  options?: WithAnySectionnableResultListOptions
```
Comment (btaillon) [inline packages/atomic/cypress/integration/result-list/result-list-utils.ts:21]: I wanted to limit `it` blocks to only relevant ones, especially given that each `it` blocks count towards Cypress' limit.

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**